### PR TITLE
Add and enforce copyright headers for C# files under /build, /src, and /tools

### PR DESCRIPTION
**Pull request by zacharycmontoya**
Changes proposed in this pull request:
- Modifies stylecop rules to enforce copyright headers on C# files
- Runs the automated rule to add the missing copyright headers
- Affects all subdirectories EXCEPT /test, which currently doesn't have stylecop applied

@DataDog/apm-dotnet

### CODE REVIEW

File: test/benchmarks/Benchmarks.Trace/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (lucaspimentel) [inline test/benchmarks/Benchmarks.Trace/Properties/launchSettings.json:1]: VS keeps trying to add this file on my machine and I keep deleting it 😅 
@kevingosse are we ok adding this?

File: stylecop.json [hunk 2]
```diff
@@ -6,6 +6,11 @@
     },
     "documentationRules": {
       "companyName": "Datadog",
+      "copyrightText": "Unless explicitly stated otherwise all files in this repository are licensed under the {licenseName} License.\nThis product includes software developed at Datadog (https://www.datadoghq.com/). Copyright {year} Datadog, Inc.",
```
Comment (lucaspimentel) [inline stylecop.json:9]: Is this the only config change and everything else is just adding the headers?

File: test/benchmarks/Benchmarks.Trace/Properties/launchSettings.json (hunk 1 above)
Comment (zacharycmontoya) [reply test/benchmarks/Benchmarks.Trace/Properties/launchSettings.json]: I had @kevingosse remove this and tell VS to stop generating it through the csproj. We should no longer have these `launchSettings.json` popping up now after rebasing off `master`

Comment (lucaspimentel) [reply test/benchmarks/Benchmarks.Trace/Properties/launchSettings.json]: thanks!

**Review by lucaspimentel — APPROVED** [review]

**Pull request merged**